docs/Contributing/architecture/software/software-overview.md

```diff
@@ -0,0 +1,70 @@
+# Software Architecture Overview
```

Comment:
This feels like a superset of the readme. We should have information in as close to one place as possible to avoid drift, so the superset here doesn't feel right. Better to link directly from the readme to the individual articles...which this article doesn't even do.

Reply:
This is the one I'll review with Tim to consider combining pages.